lib/src/camera.dart

```diff
@@ -55,7 +57,10 @@ class CameraManager {
   /// 
   /// reset the timer for FPS if needed, change resolution, enable or disable
   void updateDetails({required CameraDetails details}){  
-    this.details = details;
+    for(int i = 1; i <= 6; i++){ // [CameraDetails] has fields [1, 6]
+      this.details.setField(i, details.getField(i) ?? this.details.getField(i)); 
+    }
+    //this.details = details;
```

Comment:
I'd rather not rely on tricks like this if possible. Does `updateFromMessage` not work here? It looks like it'll do the same thing.

Reply:
i forgot the reason y updateFromMessage didnt work
but you don't think it would make sense to just details.field = newField instead of calling updateDetails() when only one field is changing